src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -371,14 +375,19 @@ void GZBridge::imuCallback(const ignition::msgs::IMU &imu)
 
 	// publish gyro
 	sensor_gyro_s sensor_gyro{};
+#if defined(ENABLE_LOCKSTEP_SCHEDULER)
 	sensor_gyro.timestamp_sample = time_us;
+	sensor_gyro.timestamp = time_us;
+#else
+	sensor_gyro.timestamp_sample = hrt_absolute_time();
+	sensor_gyro.timestamp = hrt_absolute_time();
```

Comment:
The timestamp can always be `hrt_absolute_time()`, and ideally set right before the publication. I think this must have been historical before each ignition subscription could update the clock.

Would this work everywhere?
 - `time_us` is set from the header timestamp if ENABLE_LOCKSTEP_SCHEDULER, otherwise set to hrt_absolute_time()?
 - publications always uses `time_us` for timetsamp_sample
 - always sets timestamp = hrt_absolute_time() as the last thing you do before publish (timestamp is publication metadata) 

That would hopefully keep the ifdefs to a minimum.

Reply:
I'll give it a try but I have also to handle the specific case of dt computation. Which uses time_us.